snp_application/src/bt/snp_bt_ros_nodes.cpp

```diff
@@ -154,15 +154,15 @@ BT::NodeStatus StartReconstructionServiceNode::onResponseReceived(const typename
 
 bool StopReconstructionServiceNode::setRequest(typename Request::SharedPtr& request)
 {
-  request->archive_directory = "";
+  request->archive_directory = get_parameter_or<std::string>(node_, IR_ARCHIVE_DIR_PARAM, "");
   request->mesh_filepath = get_parameter<std::string>(node_, MESH_FILE_PARAM);
-  request->min_num_faces = 1000;
+  request->min_num_faces = get_parameter<int>(node_, IR_MIN_FACES_PARAM);
 
   industrial_reconstruction_msgs::msg::NormalFilterParams norm_filt;
-  norm_filt.normal_direction.x = 0;
-  norm_filt.normal_direction.y = 0;
-  norm_filt.normal_direction.z = 1;
-  norm_filt.angle = 85;
+  norm_filt.normal_direction.x = get_parameter_or<double>(node_, IR_NORMAL_X_PARAM, 0.0);
+  norm_filt.normal_direction.y = get_parameter_or<double>(node_, IR_NORMAL_Y_PARAM, 0.0);
+  norm_filt.normal_direction.z = get_parameter_or<double>(node_, IR_NORMAL_Z_PARAM, 1.0);
+  norm_filt.angle = get_parameter_or<double>(node_, IR_NORMAL_ANGLE_TOL_PARAM, 180.0);
```

Comment:
Why are these using the `get_paramter_or` method as opposed to the `get_parameter` method?

Reply:
I want to make normal filters optional since they're not necessarily needed in all cases. I think I need to revise this approach though. Maybe if the angle is negative we don't add the filter?

Reply:
I think since you declared the parameters with a default value they will always have a value. The expected behavior in industrial_reconstruction is ["**normal_filters:** (optional) A vector of filters applied based off triangle normals, an empty vector will apply no normal filtering"](https://github.com/ros-industrial/industrial_reconstruction/tree/main?tab=readme-ov-file#stopreconstruction). So whatever you make the default you should make it default to not populate. I think what you have now with checking angle greater than 0 makes sense.

Reply:
> So whatever you make the default you should make it default to not populate

How do we do that? I thought we specified default values to maintain compatibility with Foxy and Humble

Reply:
I meant default it so that you don't push back onto the vector of filters. You appear to have set it up that way